### DISCUSSION TITLE
Patient name extraction using MedSpacy

### DISCUSSION THREAD

**Discussion by Aj-232425** [medspacy/medspacy#157] [Q&A]
Hi all,
It was nice repo to go through. But as of now i have something different requirement. What i need is to extract patient name from medical reports. Text format that i have be like : "List of encounters for: Warner, David (DOB ddmmyy) has been given Tdap/pcm . Done precautionary test to the patient: Warner, David." This is the sample text. I want to extract patient name Warner, David both the times(regardless how many times repeat ) from the page. It have previous tokens as you can see "patient" or "list of encounters" etc. How can I do this patient name extraction using MedSpacy. Any help would be appreciated.
Thanks & regards

**Comment by burgersmoke** [comment]
A very interesting application.  There certainly is not anything like this out of the box with medspacy.  

I cannot promise that I can help much with approaching this besides some brainstorming, but I find it interesting to discuss NLP tasks like this.  Can you describe a bit more about your dataset.  I don't believe I've heard of a clinical dataset where the notes are available, but patient information such as name, identifiers, demographics, etc are not available. I ask this since I'm curious about the application and objective.

Do you happen to have training data where names have already been annotated?  If so, I might be inclined to train a machine learning model.  I say this, since I would be worried that trying to solve this problem with developing rules might be very a huge effort.

All this being said, if the use cases you have do not have a significant amount of variance in their text, you might be able to try writing rules for this, but I don't think we have anything in medspacy right now that would help you retrieve patient name from text like "List of encounters for: LASTNAME, FIRSTNAME".

**Reply by patrickthealba** [reply]
Aside from hard coded rules like @burgersmoke mentioned, I would only add that you might have some help with models already released as well. I can't speak for their accuracy on your data, but Spacys en models identify person entities which includes names, and it looks like hugging face has some models worth checking out. 

https://huggingface.co/opensource/extract_names?text=I+was+looking+over+there%2C+but+Patrick+told+me+not+to.

**Comment by burgersmoke** [comment]
Oh yea.  Good call about the off-the-shelf spacy models which have entity labels for `PERSON` among other things.  I wonder how well they might work for clinical data.  Might be a good starting point.

Then after that, I wonder if you could augment whatever training data spacy uses for its NER (honestly I cannot remember which dataset(s) they use for training data) with some more clinical examples and then see how much better it can get.  Seems like this good be a good approach.

**Comment by jianlins** [comment]
https://github.com/nedap/deidentify
https://github.com/obi-ml-public/ehr_deidentification
These two might work better than generic spaCy NER.
